src/Microsoft.Azure.SignalR.Management/ServiceHubContext.cs

```diff
@@ -25,6 +25,16 @@ internal abstract class ServiceHubContext : IServiceHubContext
         /// <returns>A negotiation response object that contains an endpoint url and an access token for the client to connect to the Azure SignalR instance. </returns>
         public virtual Task<NegotiationResponse> NegotiateAsync(NegotiationOptions negotiationOptions = null, CancellationToken cancellationToken = default) => null;
 
+        /// <summary>
+        /// Close a connection asynchronously.
+        /// </summary>
+        /// <param name="connectionId">The connection to close</param>
+        /// <param name="reason">The reason to close the connection. </param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>The created <see cref="System.Threading.Tasks.Task{TResult}">Task</see> that represents the asynchronous operation.</returns>
+        /// <remarks>To get the <paramref name="reason"/> from connection closed event, client should set <see cref="NegotiationOptions.EnableDetailedErrors"/> during negotiation.</remarks>
+        public virtual Task CloseConnectionAsync(string connectionId, string reason = null, CancellationToken cancellationToken = default) => null;
```

Comment:
how about putting this into an `IServiceManager` as similar to `IGroupManager` and `IUserGroupManager`?

Reply:
There are another three methods to be added: check connection exists, check user exists, check group exists. Should we avoid interface regarding to extensibility? And the naming seems to be a problem...

Reply:
hm... check group exists fits more into `IGroupManager` though...